File: Frontend/src/interfaces/v12/design/design.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface DesignDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
}

export interface DesignCreateUpdateDto {
  name: string;
}

export interface DesignSearchDto extends PagedRequestAbstractDto {
  search: string;
}

File: Wms.Api/Dto/Design/Design.cs
 using Wms.Api.Entities; 

namespace Wms.Api.Dto.Design
{
    public class DesignDetailsDto : CreatedChangedEntity
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
    }

    public class DesignCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class DesignSearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
}
